local-cli/install/install.js

```diff
@@ -10,7 +11,17 @@ log.heading = 'rnpm-install';
 function install(args, config) {
   const name = args[0];
 
-  var res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
+  const projectDir = config.getProjectRoots()[0];
+  const isYarnAvailable =
+    yarn.getYarnVersionIfAvailable() &&
+    yarn.isGlobalCliUsingYarn(projectDir);
+
+  var res;
+  if (isYarnAvailable) {
+    res = spawnSync('yarn', ['add', name], spawnOpts);
+  } else {
+    res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
```

Comment:
I'm thinking as well but still have some questions: 

- How to handle the different command of `yarn [add]` and `npm [install --save]` ?
- Should it accept an option to allows to force use the `npm` client (like in `react-native-cli/index`)?

What if I make a function call look like:

```js
callYarnOrNpm('add some-package', 'install some-package --save', options);
```

and let it decides which command will be executed.